tests/src/test/scala/cats/tests/TraverseTests.scala

```diff
@@ -0,0 +1,53 @@
+package cats
+package tests
+
+import org.scalatest.prop.PropertyChecks
+import org.scalacheck.Arbitrary
+
+import cats.instances.all._
+
+abstract class TraverseCheck[F[_]: Traverse](name: String)(implicit ArbFInt: Arbitrary[F[Int]]) extends CatsSuite with PropertyChecks {
+
+  test(s"Traverse[$name].indexed") {
+    forAll { (fa: F[Int]) =>
+      fa.indexed.toList should === (fa.toList.zipWithIndex)
+    }
+  }
+
+  test(s"Traverse[$name].mapWithIndex") {
+    forAll { (fa: F[Int]) =>
+      fa.mapWithIndex((a, i) => (a, i)).toList should === (fa.toList.zipWithIndex)
+    }
+  }
+
+  test(s"Traverse[$name].traverseWithIndex") {
+    forAll { (fa: F[Int]) =>
+      fa.traverseWithIndex((a, i) => Option((a, i))).map(_.toList) should === (Option(fa.toList.zipWithIndex))
```

Comment:
Update incoming. I switched from `Option` to `Tuple2` to make the stronger test more straightforward.